### DISCUSSION TITLE
Uuid and index in database

### DISCUSSION THREAD

**Discussion by rudolfbruder** [spatie/laravel-event-sourcing#486] [Q&A]
Hi,

I am wondering whether I should create index on uuid columns in my models.

My tables with models that are projections will get large and I wonder what the performance will be when using uuid and event sourcing. Wont the find by uuid be slow when my table will have millions of records without index on this column?

On the other hand its a long string, should it be indexed?

**Comment by inmanturbo** [comment]
Typically yes the uuid column would be indexed. Or you could use HasUuids trait and use uuid as primary key, ie `$table->uuid('id')->primary()` 

Then with the HasUuid trait from Laravel you can use native `Model::find()`, and don't have do do any lifting for route model binding, relationship declaration, etc.

**Comment by inmanturbo** [comment]
Just make you make your id column fillable, or use `forceFill()` to write the id.
